silnlp/nmt/translate.py

```diff
@@ -1,31 +1,58 @@
 import argparse
+import os
 import time
 from pathlib import Path
-from typing import Iterable, Optional
+from typing import Iterable, Optional, Dict
+from dataclasses import dataclass
 
 import tensorflow as tf
 from machine.scripture import book_id_to_number
 
+from ..common.clearml import SILClearML
+from ..common.environment import SIL_NLP_ENV
 from ..common.paratext import book_file_name_digits
 from ..common.translator import Translator
 from ..common.utils import get_git_revision_hash
-from .config import Config, create_runner, load_config
+from .config import Config, create_runner, load_config, get_checkpoint_path
 from .utils import decode_sp_lines, encode_sp, get_best_model_dir, get_last_checkpoint
 
 
+@dataclass
 class NMTTranslator(Translator):
```

Comment:
I feel like this is overloading the `NMTTranslator` class with too many responsibilities. I think it would be better to create a separate class, maybe called `TranslationTask`, that contains an `NMTTranslator` and handles ClearML, config, and checkpoint processing.

Reply:
Hmm - if I do that, NMTTranslator becomes an almost blank class.  How about rename NMTTranslator to NMTTranslationTask and call it a day?  It is only called here.
The real updates were:
* Putting the processing code in main() into the class
* Adding 3 loc for ClearML
* Adding 4 loc for S3 buckets

If I do split it up, no one would call the NMTTranslator.  Moreover, what would be in the TranslationTask function?  Just the ClearML init?

Reply:
I think I see now - make a separate class that has as a variable the translator that can be NMT or Google.